src/main/java/org/opensearch/flowframework/transport/GetWorkflowStateAction.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.transport;
+
+import org.opensearch.action.ActionType;
+
+import static org.opensearch.flowframework.common.CommonValue.TRANSPORT_ACTION_NAME_PREFIX;
+
+/**
+ * External Action for public facing RestGetWorkflowStateAction
+ */
+public class GetWorkflowStateAction extends ActionType<GetWorkflowStateResponse> {
+    /** The name of this action */
+    public static final String NAME = TRANSPORT_ACTION_NAME_PREFIX + "workflow_state/get";
```

Comment:
nit: with different action names btw we might have a slight unintended consequence in terms of security if we also return the template body with the state later on. We might need to just add this in documentation but basically if a user has permission for workflow_state/get action but not `workflow/get` then they will still be able to get workflow with just the `workflow_state/get` action allowed, not too big of an issue I think but just calling it out